aimmo-game/tests/functional/test_movements_in_map.py

```diff
@@ -0,0 +1,174 @@
+from unittest import TestCase
+
+from mock_world import MockWorld
+from simulation.location import Location
+from simulation import map_generator
+from tests.test_simulation.dummy_avatar import MoveEastDummy, MoveWestDummy, MoveNorthDummy, MoveSouthDummy, WaitDummy
+
+
+class TestMovementsInMap(TestCase):
+
+    SETTINGS = {
+        'START_HEIGHT': 50,
+        'START_WIDTH': 50,
+        'OBSTACLE_RATIO': 0,
+    }
+
+    def set_up_environment(self, dummy_list=None, location=Location(0, 0), map_generator_class=map_generator.Main):
+        """
+        Utility method for testing.
+        """
+        self.game = MockWorld(TestMovementsInMap.SETTINGS, dummy_list, map_generator_class)
+        self.game.game_state.add_avatar(1, None, location)
+        self.avatar = self.game.avatar_manager.get_avatar(1)
+
+    def set_up_and_make_movements_in_a_single_direction(self, dummy_list, no_of_movements, spawn=Location(0, 0)):
```

Comment:
i'd use nbr_of_ number_of, or even nr_of but not no_of as it sound stoo much like a negation